packages/manager/modules/emailpro/src/domain/dkim-autoconfig/emailpro-domain-dkim-autoconfig.controller.js

```diff
@@ -62,48 +83,147 @@ export default class EmailProDomainDkimAutoconfigCtrl {
+        return this.configDkim();
+      case DKIM_STATUS.ACTIVE:
+        return this.deactivateDkim();
+      case DKIM_STATUS.DISABLED:
+        return this.activateDkim();
+      default:
+        console.error('Invalid DKIM status:', this.dkimStatus);
+        return null;
+    }
+  }
+
+  activateDkim() {
+    const dkimSelectors = this.domain.dkim;
+
+    const promise = this.services.EmailProDomains.enableDkim(
       this.services.$stateParams.productId,
       this.domain.name,
-    )
-      .then((dkimSelectors) => {
-        const promises = dkimSelectors.map((dkimSelector, index) => {
-          return this.services.EmailProDomains.postDkim(
-            this.services.$stateParams.productId,
-            this.domain.name,
-            {
-              selectorName: dkimSelector,
-              autoEnableDKIM: index === 0,
-              configureDkim: true,
-            },
-          );
-        });
-        return this.services.$q.all(promises);
-      })
+      dkimSelectors[0].selectorName,
+    );
+
+    this.handleDkimOperationResponse(promise);
+  }
+
+  deactivateDkim() {
+    const dkimSelectors = this.domain.dkim;
+
+    const promise = this.services.EmailProDomains.disableDkim(
+      this.services.$stateParams.productId,
+      this.domain.name,
+      dkimSelectors[0].status === 'inProduction'
+        ? dkimSelectors[0].selectorName
+        : dkimSelectors[1].selectorName,
+    );
+    this.handleDkimOperationResponse(promise);
+  }
+
+  getDkimSelectorForCurrentState() {
+    return this.services.EmailProDomains.getDkimSelector(
+      this.services.$stateParams.productId,
+      this.domain.name,
+    ).catch(({ data }) => {
+      this.writeError('emailpro_tab_domain_diagnostic_dkim_error', data);
+      this.leaveDkimConfigurator();
+      return this.services.$q.reject(data);
+    });
+  }
+
+  postDkimFor(selectors) {
+    return selectors.map((dkimSelector, index) => {
+      return this.services.EmailProDomains.postDkim(
+        this.services.$stateParams.productId,
+        this.domain.name,
+        {
+          selectorName: dkimSelector,
+          autoEnableDKIM: index === 0,
+          configureDkim: !this.dkimForNoOvhCloud,
+        },
+      );
+    });
+  }
+
+  stepConfigureDkim() {
+    this.stepConfigureDkimFor('emailpro');
+  }
+
+  getTitleDependingOnStep() {
+    return this.getTitleDependingOnStepFor('emailpro');
+  }
+
+  getNextButtonDependingOnStep() {
+    return this.getNextButtonDependingOnStepFor('emailpro');
+  }
+
+  getDkimName(index) {
+    return this.getDkimNameFor('emailpro', index);
+  }
+
+  getDkimRecord(index) {
+    return this.getDkimRecordFor('emailpro', index);
+  }
+
+  configDkim() {
+    const promise = this.getDkimSelectorForCurrentState();
+    promise.then((dkimSelectors) => {
+      const promises = this.postDkimFor(dkimSelectors);
+      return this.services.$q.all(promises);
+    });
+    this.handleDkimOperationResponse(promise);
+  }
```

Comment:
**nitpick:** The function names ending with `For` does not convey the complete meaning of what it is supposed to do. Please use a meaningful names.